internal/release/cross/release_list.go

```diff
@@ -171,6 +171,24 @@ func (r *ReleaseList) GetReleasesForPromotion(sourceEnv, targetEnv *v1alpha1.Env
 	return subset, nil
 }
 
+// HasNonPromotableReleases checks if the list contains releases that cannot be promoted based
+// on the version format allowed at the target environment.
+func (r *ReleaseList) HasNonPromotableReleases(sourceEnv, targetEnv *v1alpha1.Environment) bool {
```

Comment:
I would rename the function to something more indicative of what it is checking for. `GetInvalidReleaseVersions`.

I also think it should be a `get` and not a `has` because it would be nice if the error could include the release names that were invalid:

Something like:
```
cannot promote release(s): audittrail, products, external-brokerage: target env qa does not allow pre-release versions 
```